### PR TITLE
docs(k8s): add config variables section

### PR DESCRIPTION
**Pull request by douglasduteil**

### CODE REVIEW

**Comment by douglasduteil** [comment]
Inspired by latest work on https://github.com/SocialGouv/sample-next-app/tree/master/.k8s/environements/dev

File: docs/kubernetes.md [hunk 1]
```diff
@@ -56,3 +56,109 @@ Pour accéder à votre cluster :
 Plus de détails sur l'administration kube avec k9s sur [cet article](https://opensource.com/article/20/5/kubernetes-administration).
 
 Une sonde [sentry-kubernetes](https://github.com/getsentry/sentry-kubernetes) est installée sur le cluster et permet de remonter toutes les erreurs : CronJob failed, probes... C'est une source d'information préciseuse quand quelque chose ne fonctionne pas dans vos déploiements. L'accès doit être demandé à la team SRE.
+
+## Variable d'environnement dans Kubernetes
+
+On vous recommande de passer vos variables d'environnement dans par `envFrom`.
```
Comment (revolunet) [inline docs/kubernetes.md:62]:
```suggestion
On vous recommande de récupérer vos variables d'environnement avec `envFrom`.
```

File: docs/kubernetes.md [hunk 2]
```diff
@@ -56,3 +56,109 @@ Pour accéder à votre cluster :
...
+  envFrom:
+    - configMapRef:
+        name: app-env
+    - secretRef:
+        name: app-env
+```
+
+### Variables de configuration
+
+Les variables qui configure le projet dans l'environnement déployé. Ces variables sont prédictible et ne sont pas nécéssairement cachées. Example : `NODE_ENV=PRODUCTION`
```
Comment (revolunet) [inline docs/kubernetes.md:75]:
```suggestion
Les variables qui configurent le projet dans l'environnement déployé. Ces variables sont prédictibles et non-chiffrées. Example : `NODE_ENV=PRODUCTION`
```

File: docs/kubernetes.md [hunk 3]
```diff
@@ -56,3 +56,109 @@ Pour accéder à votre cluster :
...
+  NODE_ENV: "production"
+  GRAPHQL_ENDPOINT: "http://hasura/v1/graphql"
+  ACCOUNT_MAIL_SENDER: "contact@fabrique.social.gouv.fr"
+  FRONTEND_PORT: "${PORT}"
+  PRODUCTION: "false"
+```
+
+### Variables secrets
+
+Les variables de configuration qui ne doivent pas être divulgées. Ces variables sont sensibles et doivent être offusqué. Example : `JWT_SECRET=xxxxxxx`
```
Comment (revolunet) [inline docs/kubernetes.md:95]:
```suggestion
Les variables de configuration qui ne doivent pas être divulguées. Ces variables sont sensibles et doivent être offusquées. Example : `JWT_SECRET=xxxxxxx`
```

File: docs/kubernetes.md [hunk 4]
```diff
@@ -56,3 +56,109 @@ Pour accéder à votre cluster :
...
+
+Il est recommandé d'utiliser un [_SealedSecret_ par ensemble des variables du context](https://github.com/bitnami-labs/sealed-secrets) et par environnement.
+
+L'équipe SRE est en charge de la gestion des valeurs dans le _SealedSecret_ utilisés par notre projet en dev comme en prod. Les valeurs de dev sont consultables par les développers de la startup en récupérant le _Secret_ du même nom.
```
Comment (revolunet) [inline docs/kubernetes.md:99]:
```suggestion
L'équipe SRE est en charge de la gestion des valeurs dans le _SealedSecret_ utilisés par notre projet en dev comme en prod. Les valeurs de dev sont consultables par les développeurs de la startup en récupérant le _Secret_ du même nom.
```

File: docs/kubernetes.md [hunk 5]
```diff
@@ -56,3 +56,109 @@ Pour accéder à votre cluster :
...
+    type: Opaque
+  encryptedData:
+    ACCOUNT_EMAIL_SECRET: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx==
+    HASURA_GRAPHQL_ADMIN_SECRET: yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy==
+    HASURA_GRAPHQL_JWT_SECRET: zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz==
+```
+
+## Scellé un secret dans Kubernetes
+
+Nous utilisont [_SealedSecret_](https://github.com/bitnami-labs/sealed-secrets) pour scellé les secrets.
```
Comment (revolunet) [inline docs/kubernetes.md:126]:
```suggestion
Nous utilisons [_SealedSecret_](https://github.com/bitnami-labs/sealed-secrets) pour sceller les secrets.
```

File: docs/kubernetes.md [hunk 6]
```diff
@@ -56,3 +56,109 @@ Pour accéder à votre cluster :
...
+
+```sh
+# Pour consulter les logs du sealed-secrets
+$ kubectl -n sealed-secrets-system logs $(kubectl -n sealed-secrets-system get pod --no-headers -o custom-columns=":metadata.name")
+```
+
+### Scellé un secret du projet
```
Comment (revolunet) [inline docs/kubernetes.md:133]:
```suggestion
### Sceller un secret du projet
```

File: docs/kubernetes.md [hunk 7]
```diff
@@ -56,3 +56,109 @@ Pour accéder à votre cluster :
...
+
+Les secrets générer par le code pour sécuriser des communications entre services par example.
```
Comment (revolunet) [inline docs/kubernetes.md:135]:
```suggestion
Les secrets générés par le code pour sécuriser des communications entre services par exemple.
```

**Comment by revolunet** [comment]
> Inspired by latest work on https://github.com/SocialGouv/sample-next-app/tree/master/.k8s/environements/dev

s/environements/environments

**Comment by revolunet** [comment]
cool merci

File: docs/kubernetes.md [hunk 8]
```diff
@@ -56,3 +56,109 @@ Pour accéder à votre cluster :
 Plus de détails sur l'administration kube avec k9s sur [cet article](https://opensource.com/article/20/5/kubernetes-administration).
 
 Une sonde [sentry-kubernetes](https://github.com/getsentry/sentry-kubernetes) est installée sur le cluster et permet de remonter toutes les erreurs : CronJob failed, probes... C'est une source d'information préciseuse quand quelque chose ne fonctionne pas dans vos déploiements. L'accès doit être demandé à la team SRE.
+
+## Variable d'environnement dans Kubernetes
+
+On vous recommande de récupérer vos variables d'environnement avec `envFrom`.
```
Comment (revolunet) [inline docs/kubernetes.md:62]:
```suggestion
On vous recommande de récupérer vos variables d'environnement dans vos composants avec `envFrom`. Ceci permet de récupérer directement toutes les variables contenues dans une ConfigMap et/ou un Sealed-Secret. 
```

File: docs/kubernetes.md [hunk 9]
```diff
@@ -56,3 +56,109 @@ Pour accéder à votre cluster :
...
+        name: app-env
+    - secretRef:
+        name: app-env
+```
+
+### Variables de configuration
+
+Les variables qui configurent le projet dans l'environnement déployé. Ces variables sont prédictibles et non-chiffrées. Example : `NODE_ENV=PRODUCTION`
+
+Il est recommandé d'utiliser une [_ConfigMap_ par ensemble des variables du context](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#configure-all-key-value-pairs-in-a-configmap-as-container-environment-variables) et par environnement.
```
Comment (revolunet) [inline docs/kubernetes.md:77]:
```suggestion
Il est recommandé d'utiliser une [_ConfigMap_ par container](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#configure-all-key-value-pairs-in-a-configmap-as-container-environment-variables) et par environnement.
```

File: docs/kubernetes.md [hunk 10]
```diff
@@ -56,3 +56,109 @@ Pour accéder à votre cluster :
...
+  NODE_ENV: "production"
+  GRAPHQL_ENDPOINT: "http://hasura/v1/graphql"
+  ACCOUNT_MAIL_SENDER: "contact@fabrique.social.gouv.fr"
+  FRONTEND_PORT: "${PORT}"
+  PRODUCTION: "false"
+```
+
+### Variables secrets
+
+Les variables de configuration qui ne doivent pas être divulguées. Ces variables sont sensibles et doivent être offusquées. Example : `JWT_SECRET=xxxxxxx`
```
Comment (revolunet) [inline docs/kubernetes.md:95]:
```suggestion
Les variables de configuration secretes qui doivent être chifrées. Example : `JWT_SECRET=xxxxxxx`
```

File: docs/kubernetes.md [hunk 11]
```diff
@@ -56,3 +56,109 @@ Pour accéder à votre cluster :
...
+
+Il est recommandé d'utiliser un [_SealedSecret_ par ensemble des variables du context](https://github.com/bitnami-labs/sealed-secrets) et par environnement.
```
Comment (revolunet) [inline docs/kubernetes.md:97]:
```suggestion
Il est recommandé d'utiliser un [_SealedSecret_ par container](https://github.com/bitnami-labs/sealed-secrets) et par environnement.
```

File: docs/kubernetes.md [hunk 12]
```diff
@@ -56,3 +56,109 @@ Pour accéder à votre cluster :
...
+      annotations:
+        sealedsecrets.bitnami.com/cluster-wide: 'true'
+    type: Opaque
+  encryptedData:
+    ACCOUNT_EMAIL_SECRET: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx==
+    HASURA_GRAPHQL_ADMIN_SECRET: yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy==
+    HASURA_GRAPHQL_JWT_SECRET: zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz==
+```
+
+## Scellé un secret dans Kubernetes
```
Comment (revolunet) [inline docs/kubernetes.md:124]:
```suggestion
## Sceller un secret dans Kubernetes
```

File: docs/kubernetes.md [hunk 13]
```diff
@@ -56,3 +56,109 @@ Pour accéder à votre cluster :
...
+$ printf '{"type":"HS256","key": "'$(gpg --gen-random --armor 1 512)'"}' \
+  | kubectl create secret generic app-env -o yaml --dry-run=client --from-file=HASURA_GRAPHQL_JWT_SECRET=/dev/stdin \
+  | kubeseal -o yaml --scope cluster-wide --merge-into .k8s/environements/prod/hasura-env.sealed-secret.yaml
+# Scelle le mot de passe admin d'hasura par example
+$ gpg --gen-random --armor 1 128 \
+  | kubectl create secret generic app-env -o yaml --dry-run=client --from-file=HASURA_GRAPHQL_ADMIN_SECRET=/dev/stdin \
+  | kubeseal -o yaml --scope cluster-wide --merge-into .k8s/environements/dev/hasura-env.sealed-secret.yaml
+```
+
+### Scellé un secret d'infra
```
Comment (revolunet) [inline docs/kubernetes.md:152]:
```suggestion
### Sceller un secret d'infra
```

File: docs/kubernetes.md [hunk 14]
```diff
@@ -56,3 +56,109 @@ Pour accéder à votre cluster :
...
+
+```sh
+# Assurez vous de communiquer avec le bon cluster
+$ export KUBECONFIG=~/.kube/config-dev2
+# Assurez vous d'avoir le certificat correspondant au cluster
+$ export SEALED_SECRETS_CERT=https://kubeseal.dev2.fabrique.social.gouv.fr/v1/cert.pem
+# Scelle le secret jwt d'hasura par example
+$ printf '{"type":"HS256","key": "'$(gpg --gen-random --armor 1 512)'"}' \
+  | kubectl create secret generic app-env -o yaml --dry-run=client --from-file=HASURA_GRAPHQL_JWT_SECRET=/dev/stdin \
+  | kubeseal -o yaml --scope cluster-wide --merge-into .k8s/environements/prod/hasura-env.sealed-secret.yaml
```
Comment (revolunet) [inline docs/kubernetes.md:145]: `cluster-wide` in production ?

File: docs/kubernetes.md [hunk 15]
```diff
@@ -56,3 +56,109 @@ Pour accéder à votre cluster :
...
+# Scelle le mot de passe admin d'hasura par example
+$ gpg --gen-random --armor 1 128 \
+  | kubectl create secret generic app-env -o yaml --dry-run=client --from-file=HASURA_GRAPHQL_ADMIN_SECRET=/dev/stdin \
+  | kubeseal -o yaml --scope cluster-wide --merge-into .k8s/environements/dev/hasura-env.sealed-secret.yaml
+```
+
+### Scellé un secret d'infra
+
+Les secrets générer par l'infra pour accéder à des services comme un base de données managées par example.
```
Comment (revolunet) [inline docs/kubernetes.md:154]:
```suggestion
Les secrets générés par l'infra pour accéder à des services comme une base de données managée par exemple.
```

File: docs/kubernetes.md [hunk 16]
```diff
@@ -56,3 +56,109 @@ Pour accéder à votre cluster :
...
+```yaml
+# [...]
+  envFrom:
+    - configMapRef:
+        name: app-env
+    - secretRef:
+        name: app-env
+```
+
+### Variables de configuration
```
Comment (revolunet) [inline docs/kubernetes.md:73]:
```suggestion
### ConfigMap : Variables de configuration
```

File: docs/kubernetes.md [hunk 17]
```diff
@@ -56,3 +56,109 @@ Pour accéder à votre cluster :
...
+  name: app-env
+data:
+  NODE_ENV: "production"
+  GRAPHQL_ENDPOINT: "http://hasura/v1/graphql"
+  ACCOUNT_MAIL_SENDER: "contact@fabrique.social.gouv.fr"
+  FRONTEND_PORT: "${PORT}"
+  PRODUCTION: "false"
+```
+
+### Variables secrets
```
Comment (revolunet) [inline docs/kubernetes.md:93]:
```suggestion
### Sealed-secrets : Variables secretes
```

File: docs/kubernetes.md (hunk 14 above)
Comment (douglasduteil) [inline docs/kubernetes.md:145]:
```suggestion
  | kubeseal -o yaml --merge-into .k8s/environements/prod/hasura-env.sealed-secret.yaml
```

**Comment by douglasduteil** [comment]
Je sais pas écrier déso @revolunet :sob:

File: docs/kubernetes.md (hunk 16 above)
Comment (douglasduteil) [reply docs/kubernetes.md:73]: Done in https://github.com/SocialGouv/support/commit/114e8164140e4418ae79896c7a4b3ef57c6a933d

File: docs/kubernetes.md (hunk 17 above)
Comment (douglasduteil) [reply docs/kubernetes.md:93]: Done in https://github.com/SocialGouv/support/commit/ce9385675ad930ae95632ee7dc15dc2b9685e377

**Pull request merged**